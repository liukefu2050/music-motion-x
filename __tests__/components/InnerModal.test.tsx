import { render, screen } from "@testing-library/react"
import InnerModal from "../../src/components/InnerModal"
import renderWithModalRoot from "../test-utils/renderWithModalRoot"

// const sleep = () => new Promise((resolve) => setTimeout(resolve, 2000))

describe("InnerModal", () => {
  it("worked with modal root", () => {
    const { asFragment } = renderWithModalRoot(
      <InnerModal isDynamic={false}>test</InnerModal>,
    )
    expect(screen.getByText("test")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it("worked without modal root", () => {
    const modalRoot = document.createElement("div")
    const { asFragment } = render(
      <InnerModal isDynamic={true}>test</InnerModal>,
      { container: document.body.appendChild(modalRoot) },
    )

    expect((document.body.childNodes[1] as HTMLDivElement).innerHTML).toEqual(
      "<div>test</div>",
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("will clear element when beforeunload", async () => {
    const modalRoot = document.createElement("div")
    modalRoot.id = "modal_root"
    const original = window.console.error
    window.console.error = jest.fn()

    const { unmount } = renderWithModalRoot(
      <InnerModal isDynamic={false}>test</InnerModal>,
    )

    expect(screen.getByText("test")).toBeInTheDocument()

    const beforeunload = new Event("beforeunload")

    window.dispatchEvent(beforeunload)
    expect(window.console.error).toBeCalledTimes(0)

    expect(screen.queryByText("test")).toBeFalsy()
    unmount()
    expect(window.console.error).toBeCalledTimes(1)

    window.console.error = original
  })
})
