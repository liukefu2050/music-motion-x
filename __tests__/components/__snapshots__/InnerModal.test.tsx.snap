// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InnerModal worked with modal root 1`] = `
<DocumentFragment>
  <div>
    test
  </div>
</DocumentFragment>
`;

exports[`InnerModal worked without modal root 1`] = `<DocumentFragment />`;
